ui/src/app/components/copy-modal.tsx

```diff
@@ -63,57 +66,58 @@ class CopyNotebookModal extends React.Component<Props, State> {
 
   save() {
     this.setState({ loading: true });
-
-    workspacesApi().copyNotebook(
-      this.props.fromWorkspaceNamespace,
-      this.props.fromWorkspaceName,
-      this.props.fromNotebook.name,
-      {
-        toWorkspaceName: this.state.destination.id,
-        toWorkspaceNamespace: this.state.destination.namespace,
-        newName: appendNotebookFileSuffix(this.state.newName)
-      }
-      ).then((response) => {
-        this.setState({ requestState: RequestState.SUCCESS, loading: false });
-        this.props.onCopy(response);
-      }).catch((response) => {
-        const errorMsg = response.status === 400 ?
-          'Notebook with the same name already exists in the targeted workspace.' :
+    const {saveFunction, resourceType} = this.props;
+
+    saveFunction({
+      toWorkspaceName: this.state.destination.id,
+      toWorkspaceNamespace: this.state.destination.namespace,
+      newName: this.state.newName
+    }).then((response) => {
+      this.setState({ requestState: RequestState.SUCCESS, loading: false });
+      this.props.onCopy(response);
+    }).catch((response) => {
+      const errorMsg = response.status === 409 ?
+        `${ResourceTypeDisplayNames.get(resourceType)} with the same ` +
+        `name already exists in the targeted workspace.` :
+        response.status === 404 ?
+          `${ResourceTypeDisplayNames.get(resourceType)} not found in the ` +
+            `original workspace.` :
           'An error occurred while copying. Please try again.';
 
-        this.setState({
-          errorMsg: errorMsg,
-          requestState: RequestState.ERROR,
-          loading: false
-        });
+      this.setState({
+        errorMsg: errorMsg,
+        requestState: RequestState.ERROR,
+        loading: false
       });
+    });
   }
 
   goToDestinationWorkspace() {
     navigate(
-      ['workspaces', this.state.destination.namespace, this.state.destination.id, 'notebooks']
+      [
+        'workspaces',
+        this.state.destination.namespace,
+        this.state.destination.id,
+        this.props.destinationTab
```

Comment:
can we generate this off of the `ResourceType` prop? If we can, we can reduce the number of props this component requires.

Reply:
Yes, with a map. I can build that out.